lib/services/paris.ts

```diff
@@ -143,7 +144,7 @@ export class Paris{
 	 * @returns {Observable<TResult>} An Observable of the api call's result data type
 	 */
 	apiCall<TResult = any, TInput = any>(apiCallType:ApiCallType<TResult, TInput>, input?:TInput, dataOptions:DataOptions = defaultDataOptions):Observable<TResult>{
-		const cacheKey:string = JSON.stringify(input) || "{}";
+		const cacheKey:string = hashObject(input || {});
```

Comment:
A better solution might be to allow to pass a serialization function for the cache key, so it's not tied to Paris itself. That way you fix this problem, while also improving functionality in that the user can decide to cache together different calls (if there's a param that isn't relevant to caching, for example).

Something like `getCacheKey` on the ApiCall configuration, maybe?

And of course, this avoids adding another peer dependency to Paris, which adds complexity for its users.

Reply:
I think getting this as a parameter is somewhat of an overkill, for a few reasons.
1. The way serialization works is 100% internal to Paris in this case, it's not like you (as a Paris user) ever get that cache back. You to create the cache key and never see it again. If we would to supply a whole caching layer from the outside instead, might've been more useful than just supplying the key.
  1.1. If we do pass it from the outside, it should probably be as part of the `ParisConfig`, since you'll pretty likely to pass the same implementation every time. There's no real reason to use different caching mechanisms for different entities. Moreover, it's dangerous - since different functions can return the same values for different inputs, causing a collision internally.
2. It's an `optionalDependency` - if you don't ever need caching - you don't need this.
  2.1. We can put it as a [`bundledDependencies`](https://npm.github.io/using-pkgs-docs/package-json/types/bundleddependencies.html) instead, this way it doesn't need to be part of the app's direct dependencies, but still there.

Also, I think the code above is outdated, I was originally using [`object-hash`](https://www.npmjs.com/package/object-hash) to do that, but it evaluates getters, which is an unwanted side-effect in this case. Changed to use [`json-stringify-safe`](https://www.npmjs.com/package/json-stringify-safe) instead, which is just `JSON.stringify` wrapped in a safeguard for circular dependencies. I'll check why the code above shows up and update this comment.